binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsBinderEnvironmentPostProcessor.java

```diff
@@ -16,31 +16,38 @@
 
 package org.springframework.cloud.stream.binder.kafka.streams;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.Collections;
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.env.EnvironmentPostProcessor;
 import org.springframework.core.env.ConfigurableEnvironment;
 import org.springframework.core.env.MapPropertySource;
 
 /**
- * {@link EnvironmentPostProcessor} to exclude the SendToDlqAndContinue BiFunction
- * in core Spring Cloud Function by adding it to the ineligible function definitions.
+ * {@link EnvironmentPostProcessor} to ensure the {@link SendToDlqAndContinue sendToDlqAndContinue} BiFunction
+ * is excluded in core Spring Cloud Function by adding it to the ineligible function definitions.
  *
  * @author Soby Chacko
+ * @author Chris Bono
  */
 public class KafkaStreamsBinderEnvironmentPostProcessor implements EnvironmentPostProcessor {
 
-	@Override
-	public void postProcessEnvironment(ConfigurableEnvironment environment,
-									SpringApplication application) {
-		Map<String, Object> kafkaStreamsBinderIneligibleDefns = new HashMap<>();
-		kafkaStreamsBinderIneligibleDefns.put("spring.cloud.function.ineligible-definitions",
-			"sendToDlqAndContinue");
+	/**
+	 * The bean name of the SendToDlqAndContinue function - must remain in sync w/
+	 * {@link KafkaStreamsBinderSupportAutoConfiguration#sendToDlqAndContinue()}.
+	 */
+	private static final String SEND_TO_DLQ_AND_CONTINUE_BEAN_NAME = "sendToDlqAndContinue";
```

Comment:
I pulled this up into constant to highlight the fact that this is the linkage and must be kept in lock-step.